image/image.go

```diff
@@ -789,6 +789,32 @@ func copyLocalSnapFile(snapPath, targetDir string, info *snap.Info) (dstPath str
 	return dst, osutil.CopyFile(snapPath, dst, 0)
 }
 
+// CheckBasesAndProviders checks that all bases/default-providers are part of the seed
+func CheckBasesAndProviders(snapInfos map[string]*snap.Info) []error {
+	var errs []error
+	for _, info := range snapInfos {
+		// ensure base is available
+		if info.Base != "" && info.Base != "none" {
+			if _, ok := snapInfos[info.Base]; !ok {
+				errs = append(errs, fmt.Errorf("cannot use snap %q: base %q is missing", info.InstanceName(), info.Base))
+			}
+		}
+		// ensure core is available
+		if info.Base == "" && info.SnapType == snap.TypeApp && info.InstanceName() != "snapd" {
```

Comment:
I would split this out into info.EffectiveBase()

Reply:
as long as we find at least a couple other places that can use it